influxql/call_iterator.go

```diff
@@ -438,6 +438,71 @@ func newMeanIterator(input Iterator, opt IteratorOptions) (Iterator, error) {
 	}
 }
 
+// newIncreaseIterator returns an iterator for operating on a increase() call. See #7076.
+func newIncreaseIterator(input Iterator, opt IteratorOptions) (Iterator, error) {
+	switch input := input.(type) {
+	case FloatIterator:
+		createFn := func() (FloatPointAggregator, FloatPointEmitter) {
+			fn := NewFloatSliceFuncReducer(FloatIncreaseReduceSlice)
+			return fn, fn
+		}
+		return &floatReduceFloatIterator{input: newBufFloatIterator(input), opt: opt, create: createFn}, nil
+	case IntegerIterator:
+		createFn := func() (IntegerPointAggregator, IntegerPointEmitter) {
+			fn := NewIntegerSliceFuncReducer(IntegerIncreaseReduceSlice)
+			return fn, fn
+		}
+		return &integerReduceIntegerIterator{input: newBufIntegerIterator(input), opt: opt, create: createFn}, nil
+	default:
+		return nil, fmt.Errorf("unsupported increase iterator type: %T", input)
+	}
+}
+
+// FloatIncreaseReduceSlice returns the increase value within a window.
+func FloatIncreaseReduceSlice(a []FloatPoint) []FloatPoint {
```

Comment:
For performance, I think this can be implemented using the normal reduce functions. The slice versions of these functions are generally only required if the function needs to know the entire array for some reason (such as the standard implementation of median). I don't think this is needed for `increase()`.